overlord/snapstate/storehelpers.go

```diff
@@ -613,7 +613,95 @@ func collectCurrentSnaps(snapStates map[string]*SnapState, holds map[string][]st
 	return curSnaps, nil
 }
 
+// refreshCandidates is a wrapper for refreshCandidatesCore.
+//
+// It addresses the case where the store doesn't return refresh candidates for
+// snaps with already existing monitored refresh-candidates due to inconsistent
+// store return being caused by the throttling.
+// A second request is sent for eligible snaps that might have been throttled
+// with the RevisionOptions.Scheduled option turned off.
+//
+// Note: This wrapper is a short term solution and should be removed once a better
+// solution is reached.
 func refreshCandidates(ctx context.Context, st *state.State, names []string, revOpts []*RevisionOptions, user *auth.UserState, opts *store.RefreshOptions) ([]*snap.Info, map[string]*SnapState, map[string]bool, error) {
+	var revOptsByName map[string]*RevisionOptions
+	if revOpts != nil {
+		revOptsByName = make(map[string]*RevisionOptions, len(revOpts))
+		for i, opts := range revOpts {
+			revOptsByName[names[i]] = opts
+		}
+	}
+
+	updates, stateByInstanceName, ignoreValidation, err := refreshCandidatesCore(ctx, st, names, revOpts, user, opts)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var oldHints map[string]*refreshCandidate
+	if err := st.Get("refresh-candidates", &oldHints); err != nil {
+		if errors.Is(err, &state.NoStateError{}) {
+			// do nothing
+			return updates, stateByInstanceName, ignoreValidation, nil
+		}
+
+		return nil, nil, nil, fmt.Errorf("cannot get refresh-candidates: %v", err)
+	}
+
+	var missingNames []string
+	snapStates, err := All(st)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	for name, hint := range oldHints {
+		if _, ok := snapStates[name]; !ok {
+			continue
+		}
+		if !snapStates[name].IsInstalled() {
+			continue
+		}
```

Comment:
can this happen?

Reply:
Great point, I checked and found that `pruneRefreshCandidates` is called in `discard-snap`  so it might be unnecessary. do you suggest I remove it or just keep it for sanity checking?